lib/libmbedtls/core/bignum.c

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2018, ARM Limited
+ * Copyright (C) 2019, Linaro Limited
+ */
+
+#include <assert.h>
+#include <crypto/crypto.h>
+#include <mbedtls/bignum.h>
+#include <stdlib.h>
+#include <string.h>
+#include <util.h>
+
+#define ciL		(sizeof(mbedtls_mpi_uint))	/* chars in limb  */
+#define biL		(ciL << 3)			/* bits  in limb  */
+#define BITS_TO_LIMBS(i) ((i) / biL + ((i) % biL != 0))
+
+size_t crypto_bignum_num_bytes(struct bignum *a)
+{
+	assert(a != NULL);
+	return mbedtls_mpi_size((const mbedtls_mpi *)a);
+}
+
+size_t crypto_bignum_num_bits(struct bignum *a)
+{
+	assert(a != NULL);
+	return mbedtls_mpi_bitlen((const mbedtls_mpi *)a);
+}
+
+int32_t crypto_bignum_compare(struct bignum *a, struct bignum *b)
+{
+	int ret;
+
+	assert(a != NULL);
+	assert(b != NULL);
+	ret = mbedtls_mpi_cmp_mpi((const mbedtls_mpi *)a,
+				  (const mbedtls_mpi *)b);
+	return CMP_TRILEAN(ret, 0);
+}
+
+void crypto_bignum_bn2bin(const struct bignum *from, uint8_t *to)
+{
+	size_t len;
+
+	assert(from != NULL);
+	assert(to != NULL);
+	len = crypto_bignum_num_bytes((struct bignum *)from);
+	mbedtls_mpi_write_binary((mbedtls_mpi *)from, to, len);
+}
+
+TEE_Result crypto_bignum_bin2bn(const uint8_t *from, size_t fromsize,
+			 struct bignum *to)
+{
+	assert(from != NULL);
+	assert(to != NULL);
+	if (mbedtls_mpi_read_binary((mbedtls_mpi *)to, from, fromsize) != 0)
+		return TEE_ERROR_BAD_PARAMETERS;
+	return TEE_SUCCESS;
+}
+
+void crypto_bignum_copy(struct bignum *to, const struct bignum *from)
+{
+	assert(from != NULL);
+	assert(to != NULL);
+	mbedtls_mpi_copy((mbedtls_mpi *)to, (const mbedtls_mpi *)from);
```

Comment:
maybe assert return code.

Reply:
I think `panic()` on error would be more appropriate here since the function mustn't fail